src/components/templates/PosterHall/PosterHall.tsx

```diff
@@ -12,13 +12,35 @@ import { Button } from "components/atoms/Button";
 import { PosterPreview } from "./components/PosterPreview";
 import { PosterHallSearch } from "./components/PosterHallSearch";
 
+import { useUser } from "hooks/useUser";
+import { MyPersonalizedPosters } from "types/User";
+import { PosterPageVenue, PersonalizedPoster } from "types/venues";
+import { isTruthy } from "utils/types";
+
 import "./PosterHall.scss";
 
+const prepareForMyPosters = (usersPosters: MyPersonalizedPosters) => (
+  poster: WithId<PosterPageVenue>
+): WithId<PersonalizedPoster> => {
+  return {
+    ...poster,
+    isSaved: isTruthy(
+      poster.id && usersPosters[poster.id]?.includes(poster.id)
+    ),
+  };
+};
+
 export interface PosterHallProps {
   venue: WithId<GenericVenue>;
 }
 
+export const emptyPersonalizedPosters = {};
+
 export const PosterHall: React.FC<PosterHallProps> = ({ venue }) => {
+  const { userWithId } = useUser();
+  const userPosterIds =
+    userWithId?.myPersonalizedPosters ?? emptyPersonalizedPosters;
+
```

Comment:
Instead of including this at this 'outer layer' and then drilling the data through to where it's needed, we can just use the hook directly at the 'deep layer' where we want it.

I would be removing these changes from here.

Reply:
removed in current version